x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/storage/ReactiveStorageDeciderService.java

```diff
@@ -166,6 +185,8 @@ static Optional<String> singleNoDecision(Decision decision, Predicate<Decision>
         }
     }
 
+    record BytesShardIds(long bytes, SortedSet<ShardId> shardIds) {}
```

Comment:
I find this name a bit confusing, maybe we can use something else like `ShardsSize` or something more descriptive? otherwise we might want to use a `Tuple`?

Reply:
Done!